### PR TITLE
Fixed docker deployment instructions in docs

### PR DESCRIPTION
**Pull request by LiraLemur**
Signed-off-by: Sara <lira.lemur@gmail.com>

### Description of the Change

Fixed documentation as a temporary solution to https://jira.hyperledger.org/browse/IR-50

### Benefits

Better docs

### Possible Drawbacks 

Only temporary solution - the logic of using configs in docker should be fixed later

### CODE REVIEW

File: docs/source/guides/deployment.rst [hunk 1]
```diff
@@ -125,22 +125,35 @@ If they are met, you can move forward with the following command:
...
-    # Folder with configuration files
     -v ~/Developer/iroha/example:/opt/iroha_data \
-    # Blockstore volume
-    -v blockstore:/tmp/block_store \
-    # Postgres settings
+    -v blockstore:/tmp/block_store \ 
     -e POSTGRES_HOST='some-postgres' \
     -e POSTGRES_PORT='5432' \
     -e POSTGRES_PASSWORD='mysecretpassword' \
     -e POSTGRES_USER='postgres' \
```
Comment (neewy) [inline docs/source/guides/deployment.rst]:
```suggestion
    -e POSTGRES_USER='postgres' \
     -c 'max_prepared_transactions=100' \
```

Comment (neewy) [reply docs/source/guides/deployment.rst]: This is required after #1830

**Comment by neewy** [comment]
@nickaleks please leave a suggestion with a description of  -c 'max_prepared_transactions=100' parameter: why is it needed?

File: docs/source/guides/deployment.rst (hunk 1 above)
Comment (bakhtin) [reply docs/source/guides/deployment.rst]: Will not work this way. Should be:
```suggestion
     docker run ... hyperledger/iroha:latest -c 'max_prepared_transactions=100'
```

File: docs/source/guides/deployment.rst [hunk 2]
```diff
@@ -125,22 +125,35 @@ If they are met, you can move forward with the following command:
...
-    # Node keypair name
     -e KEY='node0' \
-    # Docker network name
     --network=iroha-network \
-    hyperledger/iroha:latest
+    hyperledger/iroha:latest.
+
+A little more about what these commands mean:
+
+``-p 50051:50051 \`` -  external port
```
Comment (bakhtin) [inline docs/source/guides/deployment.rst]:
```suggestion
 ``-p 50051:50051 \`` -  exposes port 50051 to the host
```

File: docs/source/guides/deployment.rst [hunk 3]
```diff
@@ -125,22 +125,35 @@ If they are met, you can move forward with the following command:
...
     -e POSTGRES_PASSWORD='mysecretpassword' \
     -e POSTGRES_USER='postgres' \
-    # Node keypair name
     -e KEY='node0' \
-    # Docker network name
     --network=iroha-network \
-    hyperledger/iroha:latest
+    hyperledger/iroha:latest.
+
+A little more about what these commands mean:
```
Comment (bakhtin) [inline docs/source/guides/deployment.rst]:
```suggestion
A little more about what these options mean:
```

File: docs/source/guides/deployment.rst [hunk 4]
```diff
@@ -125,22 +125,35 @@ If they are met, you can move forward with the following command:
...
+
+``-p 50051:50051 \`` -  external port
+
+``-v ~/Developer/iroha/example:/opt/iroha_data \`` - folder with configuration files
```
Comment (bakhtin) [inline docs/source/guides/deployment.rst]:
```suggestion
``-v ~/Developer/iroha/example:/opt/iroha_data \`` - mounts the folder with configuration files on the host into container
```

File: docs/source/guides/deployment.rst [hunk 5]
```diff
@@ -125,22 +125,35 @@ If they are met, you can move forward with the following command:
...
+
+``-v blockstore:/tmp/block_store \`` - blockstore volume
+
+``-e POSTGRES_HOST='some-postgres' \``
+    ``-e POSTGRES_PORT='5432' \``
+    ``-e POSTGRES_PASSWORD='mysecretpassword' \``
+    ``-e POSTGRES_USER='postgres' \`` - Postgres settings. Currently you only need them for testing purposes, because at the moment they are being taken from the ``config.docker`` file. 
+
+``-e KEY='node0' \`` - node keypair name
```
Comment (bakhtin) [inline docs/source/guides/deployment.rst]: I would add that `KEY` is a base name for a key. I.e., Iroha then adds `.priv` and `.pub` file extensions to it

File: docs/source/guides/deployment.rst [hunk 6]
```diff
@@ -125,22 +125,35 @@ If they are met, you can move forward with the following command:
...
+
+``--network=iroha-network \``
+    ``hyperledger/iroha:latest`` - Docker network name and corresponding version for Iroha. Latest here will correspond with master branch.
```
Comment (bakhtin) [inline docs/source/guides/deployment.rst]:
```suggestion
     ``hyperledger/iroha:latest`` - Docker network name and corresponding version for Iroha. `latest` here will correspond with master branch. 
```

File: docs/source/guides/deployment.rst (hunk 1 above)
Comment (neewy) [reply docs/source/guides/deployment.rst]: Do you mean that is should be mentioned after image name?

**Comment by nickaleks** [comment]
I would add to "Run postgres server" a section about optional parameters.
Currently, we have max_prepared_transactions option, which enables 2 phase commit optimization for iroha.
Postgres documentation suggest setting it to a max number of connections, which is 100 by default.

File: docs/source/guides/deployment.rst (hunk 3 above)
Comment (LiraLemur) [reply docs/source/guides/deployment.rst]: At least let us call them lines? I mean, options do not sound right.

File: docs/source/guides/deployment.rst (hunk 1 above)
Comment (bakhtin) [reply docs/source/guides/deployment.rst]: Yes. This is a Postgres configuration option, not Docker

File: docs/source/guides/deployment.rst (hunk 3 above)
Comment (bakhtin) [reply docs/source/guides/deployment.rst]: You can call them lines as well. But these are the *options* to `docker run` command.

File: docs/source/guides/deployment.rst (hunk 6 above)
Comment (LiraLemur) [reply docs/source/guides/deployment.rst]: fixed

File: docs/source/guides/deployment.rst (hunk 5 above)
Comment (LiraLemur) [reply docs/source/guides/deployment.rst]: added

File: docs/source/guides/deployment.rst (hunk 4 above)
Comment (LiraLemur) [reply docs/source/guides/deployment.rst]: added and fixed

File: docs/source/guides/deployment.rst (hunk 3 above)
Comment (LiraLemur) [reply docs/source/guides/deployment.rst]: please check out the latest commit

File: docs/source/guides/deployment.rst (hunk 2 above)
Comment (LiraLemur) [reply docs/source/guides/deployment.rst]: fixed

File: docs/source/guides/deployment.rst [hunk 7]
```diff
@@ -125,22 +125,35 @@ If they are met, you can move forward with the following command:
...
     docker run --name iroha \
-    # External port
     -p 50051:50051 \
-    # Folder with configuration files
     -v ~/Developer/iroha/example:/opt/iroha_data \
-    # Blockstore volume
-    -v blockstore:/tmp/block_store \
-    # Postgres settings
+    -v blockstore:/tmp/block_store \ 
     -e POSTGRES_HOST='some-postgres' \
```
[the remark this answers is not in the document]
Comment (LiraLemur) [reply docs/source/guides/deployment.rst]: I talked to @nickaleks and he said you're right. Should be ok now. Please check

File: docs/source/guides/deployment.rst (hunk 1 above)
Comment (LiraLemur) [reply docs/source/guides/deployment.rst]: I've rearranged the lines

File: docs/source/guides/deployment.rst [hunk 8]
```diff
@@ -128,31 +128,26 @@ If they are met, you can move forward with the following command:
...
-    ``-e POSTGRES_PORT='5432' \``
-    ``-e POSTGRES_PASSWORD='mysecretpassword' \``
-    ``-e POSTGRES_USER='postgres' \`` - Postgres settings. Currently you only need them for testing purposes, because at the moment they are being taken from the ``config.docker`` file. 
-
-``-e KEY='node0' \`` - node keypair name
+``-e KEY='node0' \`` - node keypair name. KEY is a base name for a key, Iroha then adds .priv and .pub file extensions to it
 
 ``--network=iroha-network \``
-    ``hyperledger/iroha:latest`` - Docker network name and corresponding version for Iroha. Latest here will correspond with master branch. 
+    ``hyperledger/iroha:latest`` - Docker network name and corresponding version for Iroha. git diff`latest` here will correspond with master branch
```
Comment (bakhtin) [inline docs/source/guides/deployment.rst]: typo: `git diff`

Comment (LiraLemur) [reply docs/source/guides/deployment.rst]: oh my. And I wondered why I had to type that again in the terminal :D 
Fixed. Thank you @bakhtin

**Comment by neewy** [comment]
It is not said where Iroha peer takes Postgres parameters. We need to mention about that in the docs here as well. There is a section above "Run Postgresql in a container".
@LiraLemur

**Comment by LiraLemur** [comment]
> It is not said where Iroha peer takes Postgres parameters. We need to mention about that in the docs here as well. There is a section above "Run Postgresql in a container".
> @LiraLemur

Hope I've got your comment right, please check.

**Pull request closed without merging**